github/repos_rules.go

```diff
@@ -395,6 +395,24 @@ type Ruleset struct {
 	Rules        []*RepositoryRule  `json:"rules,omitempty"`
 }
 
+// rulesetNoOmitBypassActors represents a GitHub ruleset object. The struct does not omit bypassActors if the field is nil or an empty array is passed.
+type rulesetNoOmitBypassActors struct {
```

Comment:
I think that you can embed the entire `Ruleset` object and shadow just the `bypass_actors`

```golang
type rulesetNoOmitBypassActors struct {
   Ruleset
   BypassActors []*BypassActor `json:"bypass_actors"`
}
```

example: https://goplay.tools/snippet/7lHZci7LW0y

@gmlewis WDYT?

Then you can use it like so:

```golang
func (s *RepositoriesService) UpdateRulesetNoBypassActor(...rs *Ruleset...
  rsNoBypassActor := &rulesetNoOmitBypassActors{}
   if rs != nil {
      rsNoBypassActor := &rulesetNoOmitBypassActors{Ruleset: *rs}
   }
```

Reply:
Sounds great, @tomfeigin - as long as we have a unit test that demonstrates this behavior (meaning a `BypassActors=nil` still gets passed through in JSON as `"bypass_actors":[]`.